docs/modules/publisher-directmethods.md

```diff
@@ -0,0 +1,427 @@
+  >      "OpcNodes":[
+  >         {
+  >             "Id":"nsu=http://microsoft.com/Opc/OpcPlc/;s=FastUInt0"
+  >          }
+  >       ]
+  >   }
+  > ```
+  >
+  >_Response_:
+  >
+  >```json
+  > {
+  >    "status":200,
+  >      "payload":{
+  >         
+  >    }
+  > ```
+
+## UnpublishAllNodes_V1
+
+UnpublishAllNodes method enables a client to remove all the nodes from a previously configured DataSetWriter. The DataSetWriter entity will be completely removed from the configuration storage.
+
+  _Request_: follows strictly the request payload schema, the OpcNodes attribute should be excluded.
+
+  _Response_: when successful - Status 200 and an empty json (`{}`) as Payload
+
+  _Exceptions_: a response corresponding to an exception will be returned if:
+
+  - request payload contains an endpoint (DataSet) that is not present in publisher configuration
+
+  - request payload contains OpcNodes
+
+  _Example_:
+
+  > _Method Name_: `UnpublishAllNodes_V1`
+  >
+  > _Payload_:
+  >
+  > ```json
+  > {
+  >      "EndpointUrl":"opc.tcp://opcplc:50000",
+  >      "DataSetWriterGroup":"Server0",
+  >      "DataSetWriterId":"Device0",
+  >      "DataSetPublishingInterval":5000
+  > }
+  > ```
+  >
+  > _Response_:
+  >
+  > ```json
+  > [
+  >      "Unpublishing all nodes succeeded for EndpointUrl: opc.tcp://opcplc:50000/"
+  > ]
+  > ```
+
+## GetConfiguredEndpoints_V1
+
+Returns the configured endpoints (Datasets)
+
+  _Request_: {}
+
+  _Response_: list of Endpoints configured (and optional parameters).
+
+  _Exceptions_: an exception is thrown when method call returns status other than 200
+
+  _Example_:
+
+  > _Method Name_: `GetConfiguredEndpoints_V1`
+  >
+  > _Response_:
+  >
+  > ```json
+  > {
+  >      "status":200,
+  >      "payload":[
+  >         {
+  >             "EndpointUrl":"opc.tcp://opcplc:50000/",
+  >             "DataSetWriterGroup":"Server0",
+  >             "DataSetWriterId":"Device0",
+  >             "DataSetPublishingInterval":5000
+  >          },
+  >          {
+  >             "EndpointUrl":"opc.tcp://opcplc:50001/"
+  >          }
+  >       ]
+  >   }
+  > ```
+
+## GetConfiguredNodesOnEndpoint_V1
+
+Returns the nodes configured for one Endpoint (Dataset)
+
+  _Request_: contains the elements necessary to uniquely identify a Dataset. The EndpointUrl is mandatory in the request, the other attributes are optional and can be used to refine your result.
+
+  _Response_: list of OpcNodes configured for the selected Endpoint (and optional parameters).
+
+  _Exceptions_: an exception is thrown when method call returns status other than 200
+
+  _Example_:
+
+  > _Method Name_: `GetConfiguredNodesOnEndpoints_V1`
+  >
+  > _Payload_:
+  >
+  > ```json
+  > {
+  >      "EndpointUrl":"opc.tcp://192.168.100.20:50000"
+  > }
+  > ```
+  >
+  > _Response_:
+  >
+  > ```json
+  > {
+  >      "status":200,
+  >      "payload":[
+  >         {
+  >             "id":"nsu=http://microsoft.com/Opc/OpcPlc/;s=SlowUInt1",
```

Comment:
need to add opcNodes object in the response. it's added here:
[https://github.com/Azure/Industrial-IoT/pull/1595/files](https://github.com/Azure/Industrial-IoT/pull/1595/files)

Reply:
I have added a note to update this after backwards compatibility issues are fixed/merged.